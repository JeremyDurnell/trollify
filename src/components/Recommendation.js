import React from "react";
import styled from "react-emotion";

const RecommendationContainer = styled("div")`
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: auto;
  width: 100%;
  border-bottom: 3px solid #181818;
`;

const AlbumArt = styled("img")`
  height: auto;
  width: 25%;
  border-radius: 50%;
`;

const SongInfo = styled("div")`
  text-align: center;
  width: 60%;
  height: 100%;
  margin-left: 5px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  font-size: 1.25em;
`;

const TrackData = styled("span")`
  font-weight: bold;
  margin-right: 1em;
`;

const PlayButtonContainer = styled("div")`
  width: 14%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const SpotifyLink = styled("a")`
  font-weight: bold;
  text-decoration: none;
  color: #1db954;
  margin-top: 0.5em;

  &:hover: {
    opacity: 0.5;
  }
`;
const Recommendation = ({ recommendation, index, handleSelect, playing }) => (
  <RecommendationContainer>
    <AlbumArt
      src={
        recommendation.album.images[0]
          ? recommendation.album.images[0].url
          : "https://avatars1.githubusercontent.com/u/29843005?s=400&u=0fec472ecd3c19b61d7a9233b611e268e7cf8c98&v=4"
      }
      alt="album cover"
    />
    <SongInfo>
      <p>
        <TrackData>Title: </TrackData>
        {recommendation.name}
      </p>
      <p>
        <TrackData>Artist:</TrackData> {recommendation.artists[0].name}
      </p>
      <p>
        <TrackData>Popularity: </TrackData>
        {recommendation.popularity}
      </p>
      <SpotifyLink href={recommendation.external_urls.spotify} target="_blank">
        Open in Spotify
      </SpotifyLink>
    </SongInfo>
    <PlayButtonContainer onClick={e => handleSelect(index, e)}>
      <i
        className={playing ? "icon fa fa-pause" : "icon fa fa-play"}
        style={{ fontSize: "3em", cursor: "pointer", color: "#1db954" }}
      />
    </PlayButtonContainer>
  </RecommendationContainer>
);

export default Recommendation;
