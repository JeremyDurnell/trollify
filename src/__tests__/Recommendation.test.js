import React from "react";
import ReactDOM from "react-dom";
import Recommendation from "../components/Recommendation";
import renderer from "react-test-renderer";

const recommendation = {
  album: {
    images: ["testimage"]
  },
  name: "testName",
  preview_url: "testurl",
  popularity: 50,
  artists: [{ name: "artist" }],
  external_urls: {
    spotify: "testurlhere"
  }
};

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Recommendation recommendation={recommendation} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly", () => {
  const withPreview = renderer
    .create(
      <Recommendation recommendation={recommendation} previewAvailable={true} />
    )
    .toJSON();
  expect(withPreview).toMatchSnapshot();

  const withoutPreview = renderer
    .create(
      <Recommendation recommendation={recommendation} previewAvailable={null} />
    )
    .toJSON();
  expect(withoutPreview).toMatchSnapshot();

  const playing = renderer
    .create(
      <Recommendation
        recommendation={recommendation}
        previewAvailable={true}
        playing={true}
      />
    )
    .toJSON();
  expect(playing).toMatchSnapshot();

  const paused = renderer
    .create(
      <Recommendation
        recommendation={recommendation}
        previewAvailable={true}
        playing={false}
      />
    )
    .toJSON();
  expect(paused).toMatchSnapshot();
});
