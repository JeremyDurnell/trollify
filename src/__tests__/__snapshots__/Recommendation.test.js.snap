// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-q5a3uo"
>
  <img
    alt="album cover"
    className="css-ovo32k"
    src={undefined}
  />
  <div
    className="css-luo11w"
  >
    <p>
      <span
        className="css-ypobiv"
      >
        Title: 
      </span>
      testName
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Artist:
      </span>
       
      artist
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Popularity: 
      </span>
      50
    </p>
    <p>
      <a
        className="css-e3j2k9"
        href="testurlhere"
        target="_blank"
      >
        Open in Spotify
      </a>
    </p>
  </div>
  <div
    className="css-pm9axs"
    onClick={[Function]}
  >
    <i
      className="icon fa fa-play"
      style={
        Object {
          "color": "#1db954",
          "cursor": "pointer",
          "fontSize": "3em",
        }
      }
    />
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="css-q5a3uo"
>
  <img
    alt="album cover"
    className="css-ovo32k"
    src={undefined}
  />
  <div
    className="css-luo11w"
  >
    <p>
      <span
        className="css-ypobiv"
      >
        Title: 
      </span>
      testName
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Artist:
      </span>
       
      artist
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Popularity: 
      </span>
      50
    </p>
    <p>
      <a
        className="css-e3j2k9"
        href="testurlhere"
        target="_blank"
      >
        Open in Spotify
      </a>
    </p>
  </div>
  <div
    className="css-pm9axs"
    onClick={[Function]}
  >
    No preview available
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="css-q5a3uo"
>
  <img
    alt="album cover"
    className="css-ovo32k"
    src={undefined}
  />
  <div
    className="css-luo11w"
  >
    <p>
      <span
        className="css-ypobiv"
      >
        Title: 
      </span>
      testName
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Artist:
      </span>
       
      artist
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Popularity: 
      </span>
      50
    </p>
    <p>
      <a
        className="css-e3j2k9"
        href="testurlhere"
        target="_blank"
      >
        Open in Spotify
      </a>
    </p>
  </div>
  <div
    className="css-pm9axs"
    onClick={[Function]}
  >
    <i
      className="icon fa fa-pause"
      style={
        Object {
          "color": "#1db954",
          "cursor": "pointer",
          "fontSize": "3em",
        }
      }
    />
  </div>
</div>
`;

exports[`renders correctly 4`] = `
<div
  className="css-q5a3uo"
>
  <img
    alt="album cover"
    className="css-ovo32k"
    src={undefined}
  />
  <div
    className="css-luo11w"
  >
    <p>
      <span
        className="css-ypobiv"
      >
        Title: 
      </span>
      testName
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Artist:
      </span>
       
      artist
    </p>
    <p>
      <span
        className="css-ypobiv"
      >
        Popularity: 
      </span>
      50
    </p>
    <p>
      <a
        className="css-e3j2k9"
        href="testurlhere"
        target="_blank"
      >
        Open in Spotify
      </a>
    </p>
  </div>
  <div
    className="css-pm9axs"
    onClick={[Function]}
  >
    <i
      className="icon fa fa-play"
      style={
        Object {
          "color": "#1db954",
          "cursor": "pointer",
          "fontSize": "3em",
        }
      }
    />
  </div>
</div>
`;
